Assets/Plugins/HEBGraph/Scripts/UIRadialDrawer.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace HEBGraph
+{
+	public class UIRadialDrawer : MaskableGraphic
+	{
+		[SerializeField] float m_outerPointInterval;
+		[SerializeField] float m_innerPointInterval;
+		[SerializeField] float m_innerRadius;
+		[SerializeField] float m_outerRadius;
+
+		[SerializeField] List<(float start, float end)> m_gaps;
+
+		public void SetGaps(List<(float start, float end)> a_gaps)
+		{
+			m_gaps = a_gaps;
+			SetVerticesDirty();
+		}
+
+		protected override void OnPopulateMesh(VertexHelper a_vh)
+		{
+			if (m_gaps == null || m_gaps.Count == 0)
+				return;
+
+			a_vh.Clear();
+
+			List<UIVertex> verts = new List<UIVertex>(100);
+			List<int> indices = new List<int>(300);
+
+			//Add initial inner and outer vertex
+			verts.Add(GetInnerVertex(0f));
+			verts.Add(GetOuterVertex(0f));
+
+			int nextGapIndex = 0;
+			int lastInnerPointIndex = 0;
+			float nextGapTheta = m_gaps[0].start;
+			float nextInnerTheta = m_innerPointInterval;
+			float nextOuterTheta = m_outerPointInterval;
+			float theta = 0f;
+			float end = Mathf.PI * 2f;
+
+			while (true)
+			{
+				if (nextGapTheta <= nextInnerTheta && nextGapTheta <= nextOuterTheta)
+				{
+					//Create gap
+					theta = nextGapTheta;
+					verts.Add(GetInnerVertex(nextGapTheta));
+					verts.Add(GetOuterVertex(nextGapTheta));
+
+					//Close current side
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)